Refractor algorithm to eliminate set usage

nzediegwu1 opened the pull request.

Commit:
Refractor algorithm to eliminate set usage

Commit:
resolve conflict

Commit:
implement validations for parameters

*validation for number
*validation for array
*validation for undefined
*validation for object

oreoluwade reviewed (commented):
Your function seems to work well when I test it on repl.it. However, this is a nodeJS project, so it would be appropriate that there is a package.json available on this branch that shows that.

nzediegwu1 commented:
Thanks a lot, I will include it asap

Commit:
include package.json

*Reason: As recommended by LFA: since the project is a node.js project, it requires package.json to be able to launch
